### ISSUE TITLE
Produce different result compare with the author's implementation 

### ISSUE THREAD

**Issue by hongsamvo** [ptirupat/AnomalyDetection_CVPR18#16]
Hi, Thank you for your amazing work.
I have the problem when run file demo.py with some videos in dataset UCF anomaly detection dataset
I used some videos which original implementation used to demo such as RoadAccidents123_x264.mp4, RoadAccidents022_x264.mp4, RoadAccidents002_x264.mp4 
Your code produced different result compare with the original implementation https://github.com/WaqasSultani/AnomalyDetectionCVPR2018
I hope that you can help to figure out the problem

**Comment by PHDPeter** [comment]
@hongsamvo it would help a lot if you could give an example of the kind of result you are getting and the kind of result you expect to get?

**Comment by hongsamvo** [comment]
@PHDPeter Hi, this folder contains the results. The videos are result from original implementation with author's shared features. The gif images are result from this work https://drive.google.com/open?id=1AjGkg8ZEaOphJZObIopYHhIf5fCVV2UG

**Comment by peterwilson3912** [comment]
> @PHDPeter Hi, this folder contains the results. The videos are result from original implementation with author's shared features. The gif images are result from this work https://drive.google.com/open?id=1AjGkg8ZEaOphJZObIopYHhIf5fCVV2UG

Hey! Did you find the reason? May I ask does this repo work?

**Comment by hongsamvo** [comment]
@karliell I think the reason is the difference of C3D model version. This repo use tensorflow version of C3D pre-trained 1milion sport and author's implemention use C3D original model. I think training the classifier again with features from tensorflow model will produce the good result.

**Comment by hongsamvo** [comment]
@karliell This repo work in my case, but I do the experiment on another dataset.

**Comment by peterwilson3912** [comment]
> @karliell This repo work in my case, but I do the experiment on another dataset.

This code works for me.
How about the implementation [AnomalyDetectionCVPR2018](https://github.com/WaqasSultani/AnomalyDetectionCVPR2018) 

Would you provide more details about the implementation?

Thank you so much!

**Comment by hongsamvo** [comment]
@karliell That is the original implementation. The author use C3D model implemented by Caffe framework. I think the original implementation very helpful for training classifier process.

**Comment by peterwilson3912** [comment]
> @karliell That is the original implementation. The author use C3D model implemented by Caffe framework. I think the original implementation very helpful for training classifier process.

Thank you so much!! That's fantastic work. 
I'm wondering what's the next step after getting c3d features.
I'm new to related topics and the explanation in that repo is not clear.
Would you please share some experience on the implementation?

Thank you so much for your help!

**Comment by hongsamvo** [comment]
@karliell Hi, after getting c3d features, I use a interpolation function( you can find in this repo or use Save_C3DFeatures_32Segment.m from original implementation, note that I found some differences between 2 implementations) to convert features to an 2d array with shape 32x4096 (32 segment per video, with 1 segment we have a feature vector with shape 4096), with these features you can train a classifier with code from original implementation.You can read the paper to get more information.

**Comment by XUYUNYUN666** [comment]
> @PHDPeter Hi, this folder contains the results. The videos are result from original implementation with author's shared features. The gif images are result from this work https://drive.google.com/open?id=1AjGkg8ZEaOphJZObIopYHhIf5fCVV2UG

I have a question, why there are two kinds of curves, as below
![image](https://user-images.githubusercontent.com/38391161/82431616-70cbee00-9ac1-11ea-8f8d-ddd87ea6dfce.png)
![image](https://user-images.githubusercontent.com/38391161/82431681-8ccf8f80-9ac1-11ea-9f47-b7955dd580dc.png)

**Comment by hongsamvo** [comment]
Hi @XUYUNYUN666 , I am not sure but I think the difference between two graphs come from the different implementation in visualization module and how interpolation the results.

**Comment by XUYUNYUN666** [comment]
> Hi @XUYUNYUN666 , I am not sure but I think the difference between two graphs come from the different implementation in visualization module and how interpolation the results.

Thank you very much. Also, I did not find the training codes, I want to train it by myself. Can you give me some help?

**Comment by hongsamvo** [comment]
Hi @XUYUNYUN666 For training classifier, you can find the code in paper's author original implementation

**Comment by PHDPeter** [comment]
> > @PHDPeter Hi, this folder contains the results. The videos are result from original implementation with author's shared features. The gif images are result from this work https://drive.google.com/open?id=1AjGkg8ZEaOphJZObIopYHhIf5fCVV2UG
> 
> Hey! Did you find the reason? May I ask does this repo work?

Yes the code works. I do not work on this code as I am using a PyTorch version (https://github.com/ekosman/AnomalyDetectionCVPR2018-Pytorch).
However I have run this code and it works for what the athers say it will do.
